mm2src/mm2_main/src/ext_api/one_inch/errors.rs

```diff
@@ -0,0 +1,83 @@
+use coins::{eth::u256_to_big_decimal, NumConversError};
+use common::{HttpStatusCode, StatusCode};
+use enum_derives::EnumFromStringify;
+use mm2_number::BigDecimal;
+use ser_error_derive::SerializeErrorType;
+use serde::Serialize;
+use trading_api::one_inch_api::errors::ApiClientError;
+
+#[derive(Debug, Display, Serialize, SerializeErrorType, EnumFromStringify)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum ApiIntegrationRpcError {
+    #[from_stringify("coins::CoinFindError")]
+    CoinFindError(String),
+    #[display(fmt = "EVM token needed")]
+    CoinTypeError,
+    #[display(fmt = "NFT not supported")]
+    NftNotSupported,
+    #[display(fmt = "Chain not supported")]
+    ChainNotSupported,
+    #[from_stringify("coins::UnexpectedDerivationMethod")]
+    MyAddressError(String),
+    InvalidParam(String),
+    #[display(fmt = "allowance not enough for 1inch contract, available: {allowance}, needed: {amount}")]
+    OneInchAllowanceNotEnough {
+        allowance: BigDecimal,
+        amount: BigDecimal,
+    },
+    #[display(fmt = "1inch API error: {}", _0)]
+    OneInchError(ApiClientError),
+    ApiDataError(String),
+}
+
+impl HttpStatusCode for ApiIntegrationRpcError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            ApiIntegrationRpcError::CoinFindError(_)
```

Comment:
Isn't this a 404? I assume we raise this error when the coin isn't found.

Reply:
fixed